packages/editor/src/epics/contractConfiguration/updateContractConfig.epic.ts

```diff
@@ -0,0 +1,53 @@
+import { ofType } from 'redux-observable';
+import { AnyAction } from 'redux';
+import { compilerActions, panesActions, contractConfigActions } from '../../actions';
+import { withLatestFrom, switchMap, catchError } from 'rxjs/operators';
+import { IProjectItem, ProjectItemTypes } from '../../models';
+import { findItemByPath } from '../../reducers/explorerLib';
+import { empty, of } from 'rxjs';
+
+export const updateContractConfig = (action$: AnyAction, state$: any) => action$.pipe(
+    ofType(compilerActions.HANDLE_COMPILE_OUTPUT),
+    withLatestFrom(state$),
+    switchMap(([action, state]) => {
+        console.log(state$);
+        const firstKey = Object.keys(action.data.contracts)[0];
+        const secondKey = Object.keys(action.data.contracts[firstKey])[0];
+        const compilerOutputData = action.data.contracts[firstKey][secondKey].metadata;
+        const parsedCompilerOutputData = JSON.parse(compilerOutputData);
+        const constructorData = parsedCompilerOutputData.output.abi.filter((obj: { type: string; }) => obj.type === 'constructor');
+        const numberOfArgs = constructorData[0].inputs.length;
+        const dappFileItem: Nullable<IProjectItem> = findItemByPath(state.explorer.tree, ['dappfile.json'], ProjectItemTypes.File);
+        if (dappFileItem != null && dappFileItem.code != null) {
+            const code = dappFileItem.code;
+            const parsedCode = JSON.parse(code);
+            if (numberOfArgs !== parsedCode.contracts[0].args.length) {
+                overwriteArray(state$, constructorData[0].inputs, parsedCode);
+                return [panesActions.saveFile(dappFileItem.id, JSON.stringify(parsedCode, null, 4))];
+            } else {
+                return empty();
+            }
+        } else {
+            return empty();
+        }
+
+    }),
+    catchError((err: any) => {
+        console.log('Error while updating the contract configuration: ', err);
+        return of(compilerActions.HANDLE_COMPILE_OUTPUT);
+    })
+);
+
+export const overwriteArray = (state$: any, constructorOutputArray: any[], parsedDappfile: any) => {
+    const arr: any[] = [];
+    for (let el of constructorOutputArray) {
+        if (el.name === 'initMessage' && state$.value.projects.project.name === 'Hello World') {
+            el = {type: 'value', value: 'Hello World!'};
+            arr.push(el);
+        } else {
+            el = { type: 'value', value: el.name };
```

Comment:
This will always assign value type no matter the constructor type. Is there a way to check if the input type is array, account or plain value?

Reply:
The thing is, the `constructorData` has a specifically declared type, i.e it will be a string, uint256... depending on what you manually pass to it at that point and then I have to convert it.

So I suppose we would have an else if block which checks for all the different plain value types, then another else if for array and lastly it could go to an else block which will just assign the type to account? 

My question is how many different types of plain values can a constructor have? I cant find anything online, is it just uint and string? :)

Reply:
Mmm didn't realize you need to convert it aswell. I guess then we can leave it as it is :)